components/Slider/src/MDCSlider.m

```diff
@@ -84,37 +88,76 @@ - (void)commonMDCSliderInit {
   [_thumbTrack addTarget:self
                   action:@selector(thumbTrackTouchCanceled:)
         forControlEvents:UIControlEventTouchCancel];
+
+  _thumbColorsForState = [@{} mutableCopy];
+  _trackFillColorsForState = [@{} mutableCopy];
   [self addSubview:_thumbTrack];
 }
 
-#pragma mark - ThumbTrack passthrough methods
+#pragma mark - Color customization methods
 
-- (void)setTrackBackgroundColor:(UIColor *)trackBackgroundColor {
-  _thumbTrack.trackOffColor =
-      trackBackgroundColor ? trackBackgroundColor : [[self class] defaultTrackOffColor];
+- (void)setTrackFillColor:(UIColor *)fillColor forState:(UIControlState)state {
+  _trackFillColorsForState[@(state)] = fillColor;
```

Comment:
If fillColor is nil I believe this will crash.

Reply:
Keys cannot be nil but values can be (it removes the mapping from the dictionary).

```objectivec
NSMutableDictionary *dict = [@{} mutableCopy];
  dict[@(1)] = @"foo";
  NSLog(@"%@", dict); // { 1 = foo; }
  dict[@(1)] = nil;
  dict[@(2)] = nil;
  NSLog(@"%@", dict); // {}
```